gengo-bin/Cargo.toml

```diff
@@ -19,6 +19,7 @@ color = ["owo-colors", "gengo/owo-colors"]
 
 [dependencies]
 clap = { version = "4", features = ["derive", "wrap_help"] }
+gix-for-configuration-only = { package = "gix", version = "0.52.0", default-features = false, features = ["max-performance"] }
```

Comment:
This makes sure that the application uses the fastest possible configuration, whereas the library ideally uses only the features that it needs. From that point of view, even the current `max-performance-safe` might be more than necessary.

Reply:
Would it make sense for gengo to have
```toml
[features]
max-performance = ["gix/max-performance"]
```

And then `gengo-bin` could have

```toml
gengo = { ..., features = ["max-performance"] }
```

I feel like it's a bit less hacky to re-export features instead of adding a dependency that doesn't actually get used at the level it's imported.

Reply:
I thought about this, but believe it's not an option to keep forward feature toggles. Instead, the officially endorsed way is to only setup the `gix` features that are actually required, and leave all others for applications to configure.

Maybe one day `cargo` allows to set features of crates in the dependency tree without having them as direct dependency. It's absolutely up to you though.